eclipse/src/points-snap.ts

```diff
@@ -0,0 +1,182 @@
+import { AnchorProvider } from "@coral-xyz/anchor";
+import {
+  CreatePositionEvent,
+  getMarketAddress,
+  InvariantEventNames,
+  IWallet,
+  Market,
+  Network,
+  parseEvent,
+  RemovePositionEvent,
+} from "@invariant-labs/sdk-eclipse";
+import {
+  ConfirmedSignatureInfo,
+  Connection,
+  ParsedTransactionWithMeta,
+  PublicKey,
+} from "@solana/web3.js";
+import * as fs from "fs";
+
+enum EventTypes {
+  OpenPosition,
+  RemovePosition,
+}
+
+const fetchAllSignatures = async (
+  connection: Connection,
+  programId: PublicKey,
+  lastTxHash: string | undefined
+) => {
+  const allSignatures: ConfirmedSignatureInfo[] = [];
+  let beforeTxHash: string | undefined = undefined;
+  let done: boolean = false;
+
+  while (!done) {
+    const signatures = await connection.getSignaturesForAddress(
+      programId,
+      { before: beforeTxHash, until: lastTxHash },
+      "confirmed"
+    );
+
+    if (signatures.length === 0) {
+      done = true;
+      break;
+    }
+
+    allSignatures.push(...signatures);
+    if (lastTxHash === undefined) {
```

Comment:
Is this check needed?

Reply:
only in the first iteration when there is not tx hash but yes